src/app/features/users-page/users-page.component.ts

```diff
@@ -1,24 +1,63 @@
-import { Component } from '@angular/core';
+import { ChangeDetectionStrategy, Component } from '@angular/core';
 import { IUsers } from '@features/users-page/interfaces/users';
 import { getUsersPending } from '@features/users-page/store/users.actions';
 import { getUsersFromState } from '@features/users-page/store/users.selectors';
 import { Store } from '@ngrx/store';
-import { Observable } from 'rxjs';
+import { BehaviorSubject, combineLatest, Observable } from 'rxjs';
+import { map } from 'rxjs/operators';
+import { CompareFilters } from '@features/users-page/filter-bar/interfaces/compare-filters';
+
+function compareUserFilter (user: IUsers, { type, value }: CompareFilters) {
+
+    switch (type) {
+
+        case 'state':
+            return user.state === value;
+        case 'gender':
+            return user.gender === value;
+        case 'dateStart':
+            return new Date(user.dateOfBirth).getTime() > new Date(value).getTime();
+        case 'dateEnd':
+            return new Date(user.dateOfBirth).getTime() < new Date(value).getTime();
+
+    }
+
+    return false;
+
+}
 
 @Component({
+    changeDetection: ChangeDetectionStrategy.OnPush,
     selector: 'crx-users-page',
     styleUrls: ['./users-page.component.scss'],
     templateUrl: './users-page.component.html'
 })
 
 export class UsersPageComponent {
 
-    public users$: Observable<IUsers[]>;
+    private filters$ = new BehaviorSubject<CompareFilters[]>([]);
+    public filterIsOpen = false;
+    public users$: Observable<IUsers[]> = combineLatest([
+        this.store.select(getUsersFromState),
+        this.filters$
+    ]).pipe(map(([users, filters]) => filters.length ? users.filter((user) =>
+        filters.reduce((acc, filter) => acc && compareUserFilter(user, filter), true)) : users));
```

Comment:
this code is hard as hell to read and understand
do not try to be super short, it will be optimised in production
you are mixing 5 shortcats: 3 methods and 2 comparison all in 2 lines